### DISCUSSION TITLE
How to create a GUID by an external program (eg php)?

### DISCUSSION THREAD

**Discussion by wings0325** [azerothcore/azerothcore-wotlk#14505] [General]
I want to write a system in php to send equipment to players by email, which involves three tables of mail, mail_item, and item_template, but the ids of these three tables are not self-increasing. What kind of algorithm should I use to generate these The ID (GUID) of the table to ensure that it does not conflict with existing data? thanks

**Comment by michaeldelago** [comment]
I believe this page from the wiki may be what you're looking for:

https://www.azerothcore.org/wiki/how-to-obtain-free-records

**Comment by Nyeriah** [comment]
You cannot send mails through external tools (for example, by writing directly into the database). Mails must be sent by the server, in your case, through a SOAP connection that executes the send items command.

The reason for this is that mails are loaded into memory so there will be issues (like duplicated mail entries). Players also won't receive any mails until they relog, and if a duplicate entry case happens the items will be lost.

**Reply by wings0325** [reply]
thank you~
It's my fault for not reading the documentation very carefully, probably because I spent too much time learning about databases. If I knew this solution, I would definitely use it. Thank you again。
